muspinsim_config/build_file.py

```diff
@@ -430,6 +439,12 @@ def main():
     if parse_dict(parse_func_dict, mu_params, file_contents):
         sys.exit(1)
 
+    # Load and append the template if specified
+    if template_path is not None:
+        # Read the file
+        with open(template_path, encoding="utf-8") as template_file:
+            file_contents += template_file.readlines()
```

Comment:
What should happen if a user adds an interaction using the wrapper which is already present in the template from the gen tool? For example if I specified `quadrupolar 2`, rather than `hyperfine 2` in test_8?

I've done some simple simulations with two definintions for the hyperfine coupling, and it seems that whichever interaction is defined last will be used - i.e. if I defined a frequency of 1, then 0.1, the output data has a time period of 10 seconds rather than 1. And vice versa. There appears to be no indication that this is taking place in the logs written my MuSpinSim.

This is obviously a MuSpinSim issue rather than a Galaxy wrapper issue - I was adding two interactions purely using the config tool on its own, to say nothing of a user just writing it themselves. In that respect, I think we at least need to print a warning to say the previous interaction is being overwritten with the new one, if not throw an error.

There is then the question of whether we should also validate this in the config tool wrapper (and by that I mean in the python script, not the xml) - i.e. should I be able to generate an input script with two `hyperfine 2` interactions? I'd say no, but it's also a question of how much effort doing the logic here is if we're then going to it when running the simulation anyway. I suppose in a way this relates to #100 - since then we could definitely be lazy and rely on MSS throwing/warning in one central location in the code.